encoder/vaapiencoder_vp8.cpp

```diff
@@ -75,6 +75,10 @@ void VaapiEncoderVP8::resetParams()
 {
     //5 times compress ratio
     m_maxCodedbufSize = width() * height() * 3 / 2;
+    //when the resolution of video is very small such as 96x96 and buffer plus the header of vp8,
+    //the size of actual output buffer maybe be bigger the m_maxCodedbufSize.
+    //So plusing 4096 avoids the situation.
+    m_maxCodedbufSize += 0x1000;
```

Comment:
I think we need find a way to calc the padding size, not a fixed value.

2016-06-30 9:45 GMT+08:00 Xu Guangxin <notifications@github.com>:

> In encoder/vaapiencoder_vp8.cpp
> <https://github.com/01org/libyami/pull/486#discussion_r69057949>:
>
> > @@ -75,6 +75,10 @@ void VaapiEncoderVP8::resetParams()
> >  {
> >      //5 times compress ratio
> >      m_maxCodedbufSize = width() * height() * 3 / 2;
> > +    //when the resolution of video is very small such as 96x96 and buffer plus the header of vp8,
> > +    //the size of actual output buffer maybe be bigger the m_maxCodedbufSize.
> > +    //So plusing 4096 avoids the situation.
> > +    m_maxCodedbufSize += 0x1000;
>
> Hi Daniel:
> The m_maxCodedbufSize is just a guessed value. plus 4k should be a problem
> for large resolution, Since it's already very big.
> We do not know the compression ratio in driver, So we use entire frame
> size previously, But Linhai found for small resolution the header size
> maybe very large. So he plus 4k to make sure it's not overflow
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/01org/libyami/pull/486/files/d8ace19d20da9eb807147ec9cb73668de1de7868#r69057949>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AA0HYUSgnqTEoI0SjrfHObsTp1XekFoZks5qQx-kgaJpZM4I8YpG>
> .
>



-- 
=======================================
Pixelworks
Room 301-303 No. 88,Lane 887 Zuchongzhi Road, Zhangjiang Hi-tech Park,
Shanghai 201203, China
Best Regards,
Jun zhao/赵军
+++++++++++++++++++++++++++++++++++++++